velox/functions/sparksql/tests/SparkBitwiseTest.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+* Copyright (c) Facebook, Inc. and its affiliates.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#include <gmock/gmock.h>
+#include <optional>
+
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+namespace facebook::velox::functions::sparksql::test {
+using namespace facebook::velox::test;
+
+namespace {
+
+static constexpr auto kMin16 = std::numeric_limits<int16_t>::min();
+static constexpr auto kMax16 = std::numeric_limits<int16_t>::max();
+static constexpr auto kMin32 = std::numeric_limits<int32_t>::min();
+static constexpr auto kMax32 = std::numeric_limits<int32_t>::max();
+static constexpr auto kMin64 = std::numeric_limits<int64_t>::min();
+static constexpr auto kMax64 = std::numeric_limits<int64_t>::max();
+static constexpr int kMaxBits = std::numeric_limits<uint64_t>::digits;
+
+class BitwiseTest : public SparkFunctionBaseTest {
+protected:
+ template <typename T>
+ std::optional<T> bitwiseFunction(
```

Comment:
Is this used?

Reply:
This part was removed in the latest commit. Thanks!